WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController+Extensions.swift

```diff
@@ -404,6 +457,90 @@ extension NotificationsViewController
 
 
 
+// MARK: - SPBucketDelegate Methods
+//
+extension NotificationsViewController: SPBucketDelegate
+{
+    public func bucket(bucket: SPBucket!, didChangeObjectForKey key: String!, forChangeType changeType: SPBucketChangeType, memberNames: [AnyObject]!) {
+        // We're only concerned with New Notification Events
+        guard changeType == .Insert else {
+            return
+        }
+
+        // If needed, show the details only if NotificationPushMaxWait hasn't elapsed
+        if pushNotificationID == key {
+            if abs(pushNotificationDate.timeIntervalSinceNow) <= Syncing.pushMaxWait {
+                showDetailsForNotificationWithID(key)
+            }
+
+            stopWaitingForNotification()
+        }
+
+        // Mark as read immediately, if needed
+        if isViewOnScreen() == true && UIApplication.sharedApplication().applicationState == .Active {
```

Comment:
This operates as expected when the app is on the iPad in multitasking mode but not in focus, right?